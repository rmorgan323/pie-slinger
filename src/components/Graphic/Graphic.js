import React from 'react';
import { PropTypes } from 'prop-types';
import './Graphic.css';

const Graphic = ({ crust, size, toppings }) => {

  const displayToppings = () => {
    return toppings.map((topping, index) => {
      return (
        <img key={index} src={`/assets/${topping.name}-${topping.amount}.svg`} alt={topping} />
      );
    });
  };

  const displayCrust = () => {
    return <img src={`/assets/crust-${crust}.svg`} alt={`${crust} crust`} />;
  };

  const pizzaClass = (pizzaSize) => {
    return `Graphic ${pizzaSize}-pizza`;
  };

  return (
    <div className={pizzaClass(size)}>
      <img src="/assets/background.svg" alt="pizza background" />
      {displayToppings()}
      {displayCrust()}
    </div>
  );
};

export default Graphic;

Graphic.propTypes = {
  crust: PropTypes.string,
  size: PropTypes.string,
  toppings: PropTypes.array
};