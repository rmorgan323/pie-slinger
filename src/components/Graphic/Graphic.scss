.Graphic {
  margin: 10px auto;
  position: relative;

  img {
    left: 0;
    position: absolute;
    top: 0;
  }
}

.small-pizza {
  height: 133px;
  width: 200px;
}

.medium-pizza {
  height: 200px;
  width: 300px;
}

.large-pizza {
  height: 266px;
  width: 400px;
}