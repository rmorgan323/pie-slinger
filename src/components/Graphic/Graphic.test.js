import React from 'react';
import { shallow } from 'enzyme';
import Graphic from './Graphic';

const mockPizza = {
  crust: "thin",
  size: "medium", 
  toppings: [
    { name: "cheese", amount: "normal" },
    { name: "pepperoni", amount: "extra" },
    { name: "sausage", amount: "extra" }
  ]
};

describe('Graphic tests', () => {
  let renderedApp;

  beforeEach(() => {
    renderedApp = shallow(
      <Graphic 
        crust={mockPizza.crust}
        size={mockPizza.size}  
        toppings={mockPizza.toppings}
      />
    );
  });

  it('should match the snapshot', () => {
    expect(renderedApp).toMatchSnapshot();
  });
});