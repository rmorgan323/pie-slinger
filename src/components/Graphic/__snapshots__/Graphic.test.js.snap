// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Graphic tests should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Graphic
    crust="thin"
    size="medium"
    toppings={
        Array [
            Object {
              "amount": "normal",
              "name": "cheese",
            },
            Object {
              "amount": "extra",
              "name": "pepperoni",
            },
            Object {
              "amount": "extra",
              "name": "sausage",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="pizza background"
          src="/assets/background.svg"
/>,
        Array [
          <img
            alt={
                        Object {
                                    "amount": "normal",
                                    "name": "cheese",
                                  }
            }
            src="/assets/cheese-normal.svg"
/>,
          <img
            alt={
                        Object {
                                    "amount": "extra",
                                    "name": "pepperoni",
                                  }
            }
            src="/assets/pepperoni-extra.svg"
/>,
          <img
            alt={
                        Object {
                                    "amount": "extra",
                                    "name": "sausage",
                                  }
            }
            src="/assets/sausage-extra.svg"
/>,
        ],
        <img
          alt="thin crust"
          src="/assets/crust-thin.svg"
/>,
      ],
      "className": "Graphic medium-pizza",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "pizza background",
          "src": "/assets/background.svg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "host",
        "props": Object {
          "alt": Object {
            "amount": "normal",
            "name": "cheese",
          },
          "src": "/assets/cheese-normal.svg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "host",
        "props": Object {
          "alt": Object {
            "amount": "extra",
            "name": "pepperoni",
          },
          "src": "/assets/pepperoni-extra.svg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "host",
        "props": Object {
          "alt": Object {
            "amount": "extra",
            "name": "sausage",
          },
          "src": "/assets/sausage-extra.svg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "thin crust",
          "src": "/assets/crust-thin.svg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="pizza background"
            src="/assets/background.svg"
/>,
          Array [
            <img
              alt={
                            Object {
                                          "amount": "normal",
                                          "name": "cheese",
                                        }
              }
              src="/assets/cheese-normal.svg"
/>,
            <img
              alt={
                            Object {
                                          "amount": "extra",
                                          "name": "pepperoni",
                                        }
              }
              src="/assets/pepperoni-extra.svg"
/>,
            <img
              alt={
                            Object {
                                          "amount": "extra",
                                          "name": "sausage",
                                        }
              }
              src="/assets/sausage-extra.svg"
/>,
          ],
          <img
            alt="thin crust"
            src="/assets/crust-thin.svg"
/>,
        ],
        "className": "Graphic medium-pizza",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "pizza background",
            "src": "/assets/background.svg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "alt": Object {
              "amount": "normal",
              "name": "cheese",
            },
            "src": "/assets/cheese-normal.svg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "alt": Object {
              "amount": "extra",
              "name": "pepperoni",
            },
            "src": "/assets/pepperoni-extra.svg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "host",
          "props": Object {
            "alt": Object {
              "amount": "extra",
              "name": "sausage",
            },
            "src": "/assets/sausage-extra.svg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "thin crust",
            "src": "/assets/crust-thin.svg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
